Incremental build often fails since clang-formatting entire repo

@Learath2 @heinrich5991 any ideas? Scratch-build always fixes it, but quite annoying:

```
[3/110] Building CXX object CMakeFiles/engine-shared.dir/src/game/generated/protocol7.cpp.o
FAILED: CMakeFiles/engine-shared.dir/src/game/generated/protocol7.cpp.o 
/usr/bin/c++ -DCONF_DEBUG -DCONF_OPENSSL -DCONF_SQL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"15.0.5\" -DGLEW_STATIC -Isrc -I/usr/include/mysql -I/home/deen/sys/include/ -g -fdiagnostics-color=always -fstack-protector-all -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -std=gnu++11 -MD -MT CMakeFiles/engine-shared.dir/src/game/generated/protocol7.cpp.o -MF CMakeFiles/engine-shared.dir/src/game/generated/protocol7.cpp.o.d -o CMakeFiles/engine-shared.dir/src/game/generated/protocol7.cpp.o -c src/game/generated/protocol7.cpp
In file included from /usr/include/c++/10.2.0/algorithm:61,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/bits/stl_algobase.h:522:27: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  522 |     __copy_move_a(const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                           ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:522:54: error: expected unqualified-id before ‘<’ token
  522 |     __copy_move_a(const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:522:54: error: expected ‘)’ before ‘<’ token
  522 |     __copy_move_a(const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                  ~                                   ^
      |                                                      )
/usr/include/c++/10.2.0/bits/stl_algobase.h:522:54: error: expected initializer before ‘<’ token
/usr/include/c++/10.2.0/bits/stl_algobase.h:530:13: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  530 |     const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |             ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:530:40: error: expected unqualified-id before ‘<’ token
  530 |     const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |                                        ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:530:40: error: expected ‘)’ before ‘<’ token
  530 |     const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |                                        ^
      |                                        )
/usr/include/c++/10.2.0/bits/stl_algobase.h:529:18: note: to match this ‘(’
  529 |     __copy_move_a(_II, _II,
      |                  ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:530:40: error: expected initializer before ‘<’ token
  530 |     const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |                                        ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:535:7: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  535 |     ::__gnu_debug::_Safe_iterator<_OIte, _OSeq, _OCat>
      |       ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:757:11: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  757 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |           ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:757:38: error: expected unqualified-id before ‘<’ token
  757 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:757:38: error: expected ‘)’ before ‘<’ token
  757 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                                      ^
      |                                      )
/usr/include/c++/10.2.0/bits/stl_algobase.h:756:27: note: to match this ‘(’
  756 |     __copy_move_backward_a(
      |                           ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:757:38: error: expected initializer before ‘<’ token
  757 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:765:11: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  765 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |           ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:765:38: error: expected unqualified-id before ‘<’ token
  765 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:765:38: error: expected ‘)’ before ‘<’ token
  765 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |                                      ^
      |                                      )
/usr/include/c++/10.2.0/bits/stl_algobase.h:764:27: note: to match this ‘(’
  764 |     __copy_move_backward_a(_II, _II,
      |                           ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:765:38: error: expected initializer before ‘<’ token
  765 |   const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&);
      |                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:770:7: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  770 |     ::__gnu_debug::_Safe_iterator<_OIte, _OSeq, _OCat>
      |       ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:899:5: error: variable or field ‘__fill_a1’ declared void
  899 |     __fill_a1(::__gnu_cxx::__normal_iterator<_Ite, _Cont> __first,
      |     ^~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:899:17: error: ‘::__gnu_cxx’ has not been declared
  899 |     __fill_a1(::__gnu_cxx::__normal_iterator<_Ite, _Cont> __first,
      |                 ^~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:899:50: error: expected primary-expression before ‘,’ token
  899 |     __fill_a1(::__gnu_cxx::__normal_iterator<_Ite, _Cont> __first,
      |                                                  ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:899:57: error: expected primary-expression before ‘>’ token
  899 |     __fill_a1(::__gnu_cxx::__normal_iterator<_Ite, _Cont> __first,
      |                                                         ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:899:59: error: ‘__first’ was not declared in this scope
  899 |     __fill_a1(::__gnu_cxx::__normal_iterator<_Ite, _Cont> __first,
      |                                                           ^~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:900:10: error: ‘::__gnu_cxx’ has not been declared
  900 |        ::__gnu_cxx::__normal_iterator<_Ite, _Cont> __last,
      |          ^~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:900:43: error: expected primary-expression before ‘,’ token
  900 |        ::__gnu_cxx::__normal_iterator<_Ite, _Cont> __last,
      |                                           ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:900:50: error: expected primary-expression before ‘>’ token
  900 |        ::__gnu_cxx::__normal_iterator<_Ite, _Cont> __last,
      |                                                  ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:900:52: error: ‘__last’ was not declared in this scope
  900 |        ::__gnu_cxx::__normal_iterator<_Ite, _Cont> __last,
      |                                                    ^~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:901:8: error: expected primary-expression before ‘const’
  901 |        const _Tp& __value)
      |        ^~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:918:22: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
  918 |     __fill_a(const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                      ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:918:49: error: expected unqualified-id before ‘<’ token
  918 |     __fill_a(const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |                                                 ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:918:49: error: expected ‘)’ before ‘<’ token
  918 |     __fill_a(const ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>&,
      |             ~                                   ^
      |                                                 )
/usr/include/c++/10.2.0/bits/stl_algobase.h:918:49: error: expected initializer before ‘<’ token
/usr/include/c++/10.2.0/bits/stl_algobase.h:1022:7: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
 1022 |     ::__gnu_debug::_Safe_iterator<_Ite, _Seq, _Cat>
      |       ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:1171:25: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
 1171 |     __equal_aux(const ::__gnu_debug::_Safe_iterator<_II1, _Seq1, _Cat1>&,
      |                         ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:1171:52: error: expected unqualified-id before ‘<’ token
 1171 |     __equal_aux(const ::__gnu_debug::_Safe_iterator<_II1, _Seq1, _Cat1>&,
      |                                                    ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:1171:52: error: expected ‘)’ before ‘<’ token
 1171 |     __equal_aux(const ::__gnu_debug::_Safe_iterator<_II1, _Seq1, _Cat1>&,
      |                ~                                   ^
      |                                                    )
/usr/include/c++/10.2.0/bits/stl_algobase.h:1171:52: error: expected initializer before ‘<’ token
/usr/include/c++/10.2.0/bits/stl_algobase.h:1178:11: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
 1178 |   const ::__gnu_debug::_Safe_iterator<_II2, _Seq2, _Cat2>&);
      |           ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:1178:38: error: expected unqualified-id before ‘<’ token
 1178 |   const ::__gnu_debug::_Safe_iterator<_II2, _Seq2, _Cat2>&);
      |                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:1178:38: error: expected ‘)’ before ‘<’ token
 1178 |   const ::__gnu_debug::_Safe_iterator<_II2, _Seq2, _Cat2>&);
      |                                      ^
      |                                      )
/usr/include/c++/10.2.0/bits/stl_algobase.h:1177:16: note: to match this ‘(’
 1177 |     __equal_aux(_II1, _II1,
      |                ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:1178:38: error: expected initializer before ‘<’ token
 1178 |   const ::__gnu_debug::_Safe_iterator<_II2, _Seq2, _Cat2>&);
      |                                      ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:1183:25: error: ‘__gnu_debug’ in namespace ‘::’ does not name a type
 1183 |     __equal_aux(const ::__gnu_debug::_Safe_iterator<_II1, _Seq1, _Cat1>&,
      |                         ^~~~~~~~~~~
/usr/include/c++/10.2.0/bits/stl_algobase.h:1183:52: error: expected unqualified-id before ‘<’ token
 1183 |     __equal_aux(const ::__gnu_debug::_Safe_iterator<_II1, _Seq1, _Cat1>&,
      |                                                    ^
/usr/include/c++/10.2.0/bits/stl_algobase.h:1183:52: error: expected ‘)’ before ‘<’ token
 1183 |     __equal_aux(const ::__gnu_debug::_Safe_iterator<_II1, _Seq1, _Cat1>&,
      |                ~                                   ^
      |                                                    )
/usr/include/c++/10.2.0/bits/stl_algobase.h:1183:52: error: expected initializer before ‘<’ token
In file included from /usr/include/c++/10.2.0/cstdlib:77,
                 from /usr/include/c++/10.2.0/bits/stl_algo.h:59,
                 from /usr/include/c++/10.2.0/algorithm:62,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/bits/std_abs.h:52:11: error: ‘abs’ has not been declared in ‘::’
   52 |   using ::abs;
      |           ^~~
In file included from /usr/include/c++/10.2.0/bits/stl_algo.h:59,
                 from /usr/include/c++/10.2.0/algorithm:62,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/cstdlib:127:11: error: ‘div_t’ has not been declared in ‘::’
  127 |   using ::div_t;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:128:11: error: ‘ldiv_t’ has not been declared in ‘::’
  128 |   using ::ldiv_t;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:130:11: error: ‘abort’ has not been declared in ‘::’
  130 |   using ::abort;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:134:11: error: ‘atexit’ has not been declared in ‘::’
  134 |   using ::atexit;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:137:11: error: ‘at_quick_exit’ has not been declared in ‘::’
  137 |   using ::at_quick_exit;
      |           ^~~~~~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:140:11: error: ‘atof’ has not been declared in ‘::’
  140 |   using ::atof;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:141:11: error: ‘atoi’ has not been declared in ‘::’
  141 |   using ::atoi;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:142:11: error: ‘atol’ has not been declared in ‘::’
  142 |   using ::atol;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:143:11: error: ‘bsearch’ has not been declared in ‘::’
  143 |   using ::bsearch;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cstdlib:144:11: error: ‘calloc’ has not been declared in ‘::’
  144 |   using ::calloc;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:145:11: error: ‘div’ has not been declared in ‘::’
  145 |   using ::div;
      |           ^~~
/usr/include/c++/10.2.0/cstdlib:146:11: error: ‘exit’ has not been declared in ‘::’
  146 |   using ::exit;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:147:11: error: ‘free’ has not been declared in ‘::’
  147 |   using ::free;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:148:11: error: ‘getenv’ has not been declared in ‘::’
  148 |   using ::getenv;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:149:11: error: ‘labs’ has not been declared in ‘::’
  149 |   using ::labs;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:150:11: error: ‘ldiv’ has not been declared in ‘::’
  150 |   using ::ldiv;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:151:11: error: ‘malloc’ has not been declared in ‘::’
  151 |   using ::malloc;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:153:11: error: ‘mblen’ has not been declared in ‘::’
  153 |   using ::mblen;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:154:11: error: ‘mbstowcs’ has not been declared in ‘::’
  154 |   using ::mbstowcs;
      |           ^~~~~~~~
/usr/include/c++/10.2.0/cstdlib:155:11: error: ‘mbtowc’ has not been declared in ‘::’
  155 |   using ::mbtowc;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:157:11: error: ‘qsort’ has not been declared in ‘::’
  157 |   using ::qsort;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:160:11: error: ‘quick_exit’ has not been declared in ‘::’
  160 |   using ::quick_exit;
      |           ^~~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:163:11: error: ‘rand’ has not been declared in ‘::’
  163 |   using ::rand;
      |           ^~~~
/usr/include/c++/10.2.0/cstdlib:164:11: error: ‘realloc’ has not been declared in ‘::’
  164 |   using ::realloc;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cstdlib:165:11: error: ‘srand’ has not been declared in ‘::’
  165 |   using ::srand;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:166:11: error: ‘strtod’ has not been declared in ‘::’
  166 |   using ::strtod;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:167:11: error: ‘strtol’ has not been declared in ‘::’
  167 |   using ::strtol;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:168:11: error: ‘strtoul’ has not been declared in ‘::’
  168 |   using ::strtoul;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cstdlib:169:11: error: ‘system’ has not been declared in ‘::’
  169 |   using ::system;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:171:11: error: ‘wcstombs’ has not been declared in ‘::’
  171 |   using ::wcstombs;
      |           ^~~~~~~~
/usr/include/c++/10.2.0/cstdlib:172:11: error: ‘wctomb’ has not been declared in ‘::’
  172 |   using ::wctomb;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:200:11: error: ‘lldiv_t’ has not been declared in ‘::’
  200 |   using ::lldiv_t;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cstdlib:206:11: error: ‘_Exit’ has not been declared in ‘::’
  206 |   using ::_Exit;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:210:11: error: ‘llabs’ has not been declared in ‘::’
  210 |   using ::llabs;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:216:11: error: ‘lldiv’ has not been declared in ‘::’
  216 |   using ::lldiv;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:227:11: error: ‘atoll’ has not been declared in ‘::’
  227 |   using ::atoll;
      |           ^~~~~
/usr/include/c++/10.2.0/cstdlib:228:11: error: ‘strtoll’ has not been declared in ‘::’
  228 |   using ::strtoll;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cstdlib:229:11: error: ‘strtoull’ has not been declared in ‘::’
  229 |   using ::strtoull;
      |           ^~~~~~~~
/usr/include/c++/10.2.0/cstdlib:231:11: error: ‘strtof’ has not been declared in ‘::’
  231 |   using ::strtof;
      |           ^~~~~~
/usr/include/c++/10.2.0/cstdlib:232:11: error: ‘strtold’ has not been declared in ‘::’
  232 |   using ::strtold;
      |           ^~~~~~~
/usr/include/c++/10.2.0/cstdlib:240:11: error: ‘::__gnu_cxx’ has not been declared
  240 |   using ::__gnu_cxx::lldiv_t;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:242:11: error: ‘::__gnu_cxx’ has not been declared
  242 |   using ::__gnu_cxx::_Exit;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:244:11: error: ‘::__gnu_cxx’ has not been declared
  244 |   using ::__gnu_cxx::llabs;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:245:11: error: ‘::__gnu_cxx’ has not been declared
  245 |   using ::__gnu_cxx::div;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:246:11: error: ‘::__gnu_cxx’ has not been declared
  246 |   using ::__gnu_cxx::lldiv;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:248:11: error: ‘::__gnu_cxx’ has not been declared
  248 |   using ::__gnu_cxx::atoll;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:249:11: error: ‘::__gnu_cxx’ has not been declared
  249 |   using ::__gnu_cxx::strtof;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:250:11: error: ‘::__gnu_cxx’ has not been declared
  250 |   using ::__gnu_cxx::strtoll;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:251:11: error: ‘::__gnu_cxx’ has not been declared
  251 |   using ::__gnu_cxx::strtoull;
      |           ^~~~~~~~~
/usr/include/c++/10.2.0/cstdlib:252:11: error: ‘::__gnu_cxx’ has not been declared
  252 |   using ::__gnu_cxx::strtold;
      |           ^~~~~~~~~
In file included from /usr/include/c++/10.2.0/exception:147,
                 from /usr/include/c++/10.2.0/new:41,
                 from /usr/include/c++/10.2.0/bits/stl_construct.h:59,
                 from /usr/include/c++/10.2.0/bits/stl_tempbuf.h:60,
                 from /usr/include/c++/10.2.0/bits/stl_algo.h:62,
                 from /usr/include/c++/10.2.0/algorithm:62,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/bits/exception_ptr.h: In function ‘protocol7::std::__exception_ptr::exception_ptr protocol7::std::make_exception_ptr(_Ex)’:
/usr/include/c++/10.2.0/bits/exception_ptr.h:191:45: error: must ‘#include <typeinfo>’ before using ‘typeid’
   40 | #include <new>
  +++ |+#include <typeinfo>
   41 | 
......
  191 |    __e, const_cast<std::type_info*>(&typeid(__ex)),
      |                                             ^~~~
In file included from /usr/include/c++/10.2.0/bits/stl_construct.h:59,
                 from /usr/include/c++/10.2.0/bits/stl_tempbuf.h:60,
                 from /usr/include/c++/10.2.0/bits/stl_algo.h:62,
                 from /usr/include/c++/10.2.0/algorithm:62,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/new: At global scope:
/usr/include/c++/10.2.0/new:126:26: error: ‘void* protocol7::operator new(protocol7::std::size_t)’ may not be declared within a namespace
  126 | _GLIBCXX_NODISCARD void* operator new(std::size_t) _GLIBCXX_THROW (std::bad_alloc)
      |                          ^~~~~~~~
/usr/include/c++/10.2.0/new:128:26: error: ‘void* protocol7::operator new [](protocol7::std::size_t)’ may not be declared within a namespace
  128 | _GLIBCXX_NODISCARD void* operator new[](std::size_t) _GLIBCXX_THROW (std::bad_alloc)
      |                          ^~~~~~~~
/usr/include/c++/10.2.0/new:130:6: error: ‘void protocol7::operator delete(void*)’ may not be declared within a namespace
  130 | void operator delete(void*) _GLIBCXX_USE_NOEXCEPT
      |      ^~~~~~~~
/usr/include/c++/10.2.0/new:132:6: error: ‘void protocol7::operator delete [](void*)’ may not be declared within a namespace
  132 | void operator delete[](void*) _GLIBCXX_USE_NOEXCEPT
      |      ^~~~~~~~
/usr/include/c++/10.2.0/new:140:26: error: ‘void* protocol7::operator new(protocol7::std::size_t, const protocol7::std::nothrow_t&)’ may not be declared within a namespace
  140 | _GLIBCXX_NODISCARD void* operator new(std::size_t, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
      |                          ^~~~~~~~
/usr/include/c++/10.2.0/new:142:26: error: ‘void* protocol7::operator new [](protocol7::std::size_t, const protocol7::std::nothrow_t&)’ may not be declared within a namespace
  142 | _GLIBCXX_NODISCARD void* operator new[](std::size_t, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
      |                          ^~~~~~~~
/usr/include/c++/10.2.0/new:144:6: error: ‘void protocol7::operator delete(void*, const protocol7::std::nothrow_t&)’ may not be declared within a namespace
  144 | void operator delete(void*, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
      |      ^~~~~~~~
/usr/include/c++/10.2.0/new:146:6: error: ‘void protocol7::operator delete [](void*, const protocol7::std::nothrow_t&)’ may not be declared within a namespace
  146 | void operator delete[](void*, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
      |      ^~~~~~~~
/usr/include/c++/10.2.0/new:174:33: error: ‘void* protocol7::operator new(protocol7::std::size_t, void*)’ may not be declared within a namespace
  174 | _GLIBCXX_NODISCARD inline void* operator new(std::size_t, void* __p) _GLIBCXX_USE_NOEXCEPT
      |                                 ^~~~~~~~
/usr/include/c++/10.2.0/new:176:33: error: ‘void* protocol7::operator new [](protocol7::std::size_t, void*)’ may not be declared within a namespace
  176 | _GLIBCXX_NODISCARD inline void* operator new[](std::size_t, void* __p) _GLIBCXX_USE_NOEXCEPT
      |                                 ^~~~~~~~
/usr/include/c++/10.2.0/new:180:13: error: ‘void protocol7::operator delete(void*, void*)’ may not be declared within a namespace
  180 | inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
      |             ^~~~~~~~
/usr/include/c++/10.2.0/new:181:13: error: ‘void protocol7::operator delete [](void*, void*)’ may not be declared within a namespace
  181 | inline void operator delete[](void*, void*) _GLIBCXX_USE_NOEXCEPT { }
      |             ^~~~~~~~
In file included from /usr/include/c++/10.2.0/bits/stl_algo.h:62,
                 from /usr/include/c++/10.2.0/algorithm:62,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/bits/stl_tempbuf.h: In function ‘protocol7::std::pair<_Tp*, long int> protocol7::std::get_temporary_buffer(protocol7::std::ptrdiff_t)’:
/usr/include/c++/10.2.0/bits/stl_tempbuf.h:110:20: error: too many arguments to function ‘void* operator new(long unsigned int)’
  110 |        std::nothrow));
      |                    ^
In file included from /usr/include/c++/10.2.0/algorithm:62,
                 from src/base/tl/base.h:6,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/bits/stl_algo.h: In function ‘void protocol7::std::random_shuffle(_RAIter, _RAIter)’:
/usr/include/c++/10.2.0/bits/stl_algo.h:4597:13: error: ‘rand’ is not a member of ‘protocol7::std’; did you mean ‘rank’?
 4597 |      + std::rand() % ((__i - __first) + 1);
      |             ^~~~
      |             rank
In file included from src/base/system.h:18,
                 from src/base/tl/base.h:7,
                 from src/base/tl/range.h:6,
                 from src/base/tl/array.h:7,
                 from src/engine/shared/uuid_manager.h:4,
                 from src/engine/message.h:7,
                 from src/game/generated/protocol7.h:5,
                 from src/game/generated/protocol7.cpp:1:
/usr/include/c++/10.2.0/stdlib.h: At global scope:
/usr/include/c++/10.2.0/stdlib.h:38:12: error: ‘abort’ has not been declared in ‘protocol7::std’
   38 | using std::abort;
      |            ^~~~~
/usr/include/c++/10.2.0/stdlib.h:39:12: error: ‘atexit’ has not been declared in ‘protocol7::std’
   39 | using std::atexit;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:40:12: error: ‘exit’ has not been declared in ‘protocol7::std’
   40 | using std::exit;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:43:14: error: ‘at_quick_exit’ has not been declared in ‘protocol7::std’
   43 |   using std::at_quick_exit;
      |              ^~~~~~~~~~~~~
/usr/include/c++/10.2.0/stdlib.h:46:14: error: ‘quick_exit’ has not been declared in ‘protocol7::std’
   46 |   using std::quick_exit;
      |              ^~~~~~~~~~
/usr/include/c++/10.2.0/stdlib.h:51:12: error: ‘div_t’ has not been declared in ‘protocol7::std’
   51 | using std::div_t;
      |            ^~~~~
/usr/include/c++/10.2.0/stdlib.h:52:12: error: ‘ldiv_t’ has not been declared in ‘protocol7::std’
   52 | using std::ldiv_t;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:55:12: error: ‘atof’ has not been declared in ‘protocol7::std’
   55 | using std::atof;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:56:12: error: ‘atoi’ has not been declared in ‘protocol7::std’
   56 | using std::atoi;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:57:12: error: ‘atol’ has not been declared in ‘protocol7::std’
   57 | using std::atol;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:58:12: error: ‘bsearch’ has not been declared in ‘protocol7::std’
   58 | using std::bsearch;
      |            ^~~~~~~
/usr/include/c++/10.2.0/stdlib.h:59:12: error: ‘calloc’ has not been declared in ‘protocol7::std’
   59 | using std::calloc;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:61:12: error: ‘free’ has not been declared in ‘protocol7::std’
   61 | using std::free;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:62:12: error: ‘getenv’ has not been declared in ‘protocol7::std’
   62 | using std::getenv;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:63:12: error: ‘labs’ has not been declared in ‘protocol7::std’
   63 | using std::labs;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:64:12: error: ‘ldiv’ has not been declared in ‘protocol7::std’
   64 | using std::ldiv;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:65:12: error: ‘malloc’ has not been declared in ‘protocol7::std’
   65 | using std::malloc;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:67:12: error: ‘mblen’ has not been declared in ‘protocol7::std’
   67 | using std::mblen;
      |            ^~~~~
/usr/include/c++/10.2.0/stdlib.h:68:12: error: ‘mbstowcs’ has not been declared in ‘protocol7::std’
   68 | using std::mbstowcs;
      |            ^~~~~~~~
/usr/include/c++/10.2.0/stdlib.h:69:12: error: ‘mbtowc’ has not been declared in ‘protocol7::std’
   69 | using std::mbtowc;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:71:12: error: ‘qsort’ has not been declared in ‘protocol7::std’
   71 | using std::qsort;
      |            ^~~~~
/usr/include/c++/10.2.0/stdlib.h:72:12: error: ‘rand’ has not been declared in ‘protocol7::std’
   72 | using std::rand;
      |            ^~~~
/usr/include/c++/10.2.0/stdlib.h:73:12: error: ‘realloc’ has not been declared in ‘protocol7::std’
   73 | using std::realloc;
      |            ^~~~~~~
/usr/include/c++/10.2.0/stdlib.h:74:12: error: ‘srand’ has not been declared in ‘protocol7::std’
   74 | using std::srand;
      |            ^~~~~
/usr/include/c++/10.2.0/stdlib.h:75:12: error: ‘strtod’ has not been declared in ‘protocol7::std’
   75 | using std::strtod;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:76:12: error: ‘strtol’ has not been declared in ‘protocol7::std’
   76 | using std::strtol;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:77:12: error: ‘strtoul’ has not been declared in ‘protocol7::std’
   77 | using std::strtoul;
      |            ^~~~~~~
/usr/include/c++/10.2.0/stdlib.h:78:12: error: ‘system’ has not been declared in ‘protocol7::std’
   78 | using std::system;
      |            ^~~~~~
/usr/include/c++/10.2.0/stdlib.h:80:12: error: ‘wcstombs’ has not been declared in ‘protocol7::std’
   80 | using std::wcstombs;
      |            ^~~~~~~~
/usr/include/c++/10.2.0/stdlib.h:81:12: error: ‘wctomb’ has not been declared in ‘protocol7::std’
   81 | using std::wctomb;
      |            ^~~~~~
cc1plus: note: unrecognized command-line option ‘-Wno-nullability-completeness’ may have been intended to silence earlier diagnostics
[11/110] Building CXX object CMakeFiles/game-shared.dir/src/game/gamecore.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/datasrc/compile.py
+++ b/datasrc/compile.py
@@ -98,6 +98,8 @@ if gen_network_header:
 
 	print("#ifndef GAME_GENERATED_PROTOCOL_H")
 	print("#define GAME_GENERATED_PROTOCOL_H")
+	print("#include <engine/shared/protocol.h>")
+	print("#include <engine/message.h>")
 	print(network.RawHeader)
 
 	for e in network.Enums:
@@ -167,9 +169,9 @@ if gen_network_source:
 	# create names
 	lines = []
 
+	lines += ['#include "protocol.h"']
 	lines += ['#include <engine/shared/protocol.h>']
 	lines += ['#include <engine/message.h>']
-	lines += ['#include "protocol.h"']
 	lines += ['#include <game/mapitems_ex.h>']
 
 	lines += ['CNetObjHandler::CNetObjHandler()']

--- a/datasrc/seven/compile.py
+++ b/datasrc/seven/compile.py
@@ -99,6 +99,8 @@ if gen_network_header:
 
 	print("#ifndef GAME_GENERATED_PROTOCOL7_H")
 	print("#define GAME_GENERATED_PROTOCOL7_H")
+	print("#include <engine/shared/protocol.h>")
+	print("#include <engine/message.h>")
 	print("namespace protocol7 {")
 	print(network.RawHeader)
 
@@ -177,9 +179,9 @@ if gen_network_source:
 	# create names
 	lines = []
 
+	lines += ['#include "protocol7.h"']
 	lines += ['#include <engine/shared/protocol.h>']
 	lines += ['#include <engine/message.h>']
-	lines += ['#include "protocol7.h"']
 
 	lines += ['namespace protocol7 {']
 

--- a/scripts/fix_style.py
+++ b/scripts/fix_style.py
@@ -8,13 +8,20 @@ import sys
 
 os.chdir(os.path.dirname(__file__) + "/..")
 
-ignore_files = ["src/engine/keys.h", "src/engine/client/keynames.h"]
-
 def recursive_file_list(path):
 	result = []
 	for dirpath, dirnames, filenames in os.walk(path):
-		result += filter(lambda p: p not in ignore_files, [os.path.join(dirpath, filename) for filename in filenames])
+		result += [os.path.join(dirpath, filename) for filename in filenames]
 	return result
+
+IGNORE_FILES = [
+	"src/engine/client/keynames.h",
+	"src/engine/keys.h",
+]
+def filter_ignored(filenames):
+	return [filename for filename in filenames
+		if filename not in IGNORE_FILES
+		and not filename.startswith("src/game/generated/")]
 
 def filter_cpp(filenames):
 	return [filename for filename in filenames
@@ -31,7 +38,7 @@ def main():
 	p = argparse.ArgumentParser(description="Check and fix style of changed files")
 	p.add_argument("-n", "--dry-run", action="store_true", help="Don't fix, only warn")
 	args = p.parse_args()
-	filenames = filter_cpp(recursive_file_list("src"))
+	filenames = filter_ignored(filter_cpp(recursive_file_list("src")))
 	if not args.dry_run:
 		reformat(filenames)
 	else:

